### PR TITLE
feat: send CoJ signed event to SQS

### PR DESCRIPTION
**Pull request by ReubenRobertsHEE**
TIS21-4430: Event handler to push CoJ data to SQS queue

### CODE REVIEW

File: src/main/java/uk/nhs/hee/trainee/details/service/EventPublishService.java [hunk 1]
```diff
@@ -55,6 +61,21 @@ public void publishProfileCreateEvent(TraineeProfile profile) {
...
+  /**
+   * Publish a CoJ signed event.
+   *
+   * @param programmeMembership The signed {@link ProgrammeMembership}.
+   */
+  public void publishCojSignedEvent(ProgrammeMembership programmeMembership) {
+    log.info("Sending CoJ signed event for programme membership id '{}'",
+        programmeMembership.getTisId());
+
+    ConditionsOfJoining conditionsOfJoining = programmeMembership.getConditionsOfJoining();
```
Comment (ReubenRobertsHEE) [inline src/main/java/uk/nhs/hee/trainee/details/service/EventPublishService.java:76]: It is assumed that the CoJ will not be null, based on the code upstream - but we could add an additional check here if needed (and only send the message if its not null).

Comment (Judge40) [reply src/main/java/uk/nhs/hee/trainee/details/service/EventPublishService.java:76]: I think this is a fair assumption, it might also be something we want to allow to happen in case we ever need to send a null to tell TIS to "unsign" a CoJ in the future.

File: src/test/java/uk/nhs/hee/trainee/details/service/EventPublishServiceTest.java [hunk 2]
```diff
@@ -60,4 +66,29 @@ void shouldPublishProfileCreateEvent() {
...
+    verify(messagingTemplate).convertAndSend(eq(COJ_QUEUE_URL), eventCaptor.capture());
+
+    CojSignedEvent event = eventCaptor.getValue();
+    assertThat("Unexpected programme membership ID.",
+        event.getProgrammeMembershipTisId(), is("123"));
+    ConditionsOfJoining conditionsOfJoiningSent = event.getConditionsOfJoining();
+    assertThat("Unexpected CoJ Signed At",
+        conditionsOfJoining.signedAt(), is(conditionsOfJoiningSent.signedAt()));
+    assertThat("Unexpected CoJ Version",
+        conditionsOfJoining.version(), is(conditionsOfJoiningSent.version()));
```
Comment (Judge40) [inline src/test/java/uk/nhs/hee/trainee/details/service/EventPublishServiceTest.java:92]: It's a bit of an edge case, but I've check against the original values rather than pulling them back out of `conditionsOfJoiningSent`. From a test perspective we can't guarantee a lack of side effects on the passed object tricking our test in to thinking all is well.

**Review by Judge40 — CHANGES_REQUESTED** [review]

File: src/test/java/uk/nhs/hee/trainee/details/service/EventPublishServiceTest.java (hunk 2 above)
Comment (Judge40) [reply src/test/java/uk/nhs/hee/trainee/details/service/EventPublishServiceTest.java:92]: Also, getting myself confused about which var was which highlighted that were testing these two var in an odd order. We're asserting on the original object rather than the captured "sent" object.

Comment (ReubenRobertsHEE) [reply src/test/java/uk/nhs/hee/trainee/details/service/EventPublishServiceTest.java]: (Sorry, response was to your original point. Yes, I picked up on the wrong order as well and fixed it in the next commit)

**Review by Judge40 — APPROVED** [review]
